public/app/features/alerting/unified/components/receivers/editor/autocomplete.ts

```diff
@@ -0,0 +1,119 @@
+      }
+
+      if (context.triggerKind === monaco.languages.CompletionTriggerKind.Invoke && !context.triggerCharacter) {
+        return completionProvider.getFunctionsSuggestions();
+      }
+
+      const wordBeforeDot = model.getWordUntilPosition({
+        lineNumber: position.lineNumber,
+        column: position.column - 1,
+      });
+
+      return completionProvider.getTemplateDataSuggestions(wordBeforeDot.word);
+    },
+  };
+
+  return monaco.languages.registerCompletionItemProvider('go-template', goTemplateAutocompleteProvider);
+}
+
+function isInsideGoExpression(model: editor.ITextModel, position: Position) {
+  const searchRange = {
+    startLineNumber: position.lineNumber,
+    endLineNumber: position.lineNumber,
+    startColumn: model.getLineMinColumn(position.lineNumber),
+    endColumn: model.getLineMaxColumn(position.lineNumber),
+  };
+
+  const goSyntaxRegex = '\\{\\{[a-zA-Z0-9._() "]+\\}\\}';
+  const matches = model.findMatches(goSyntaxRegex, searchRange, true, false, null, true);
+
+  return matches.some((match) => match.range.containsPosition(position));
+}
+
+export class CompletionProvider {
+  constructor(private readonly monaco: Monaco, private readonly range: IRange) {}
+
+  getSnippetsSuggestions = (): languages.ProviderResult<languages.CompletionList> => {
+    return this.getCompletionsFromDefinitions(getSnippetsSuggestions(this.monaco));
+  };
+
+  getFunctionsSuggestions = (): languages.ProviderResult<languages.CompletionList> => {
+    return this.getCompletionsFromDefinitions(getAlertManagerSuggestions(this.monaco));
+  };
+
+  getTemplateDataSuggestions = (wordContext: string): languages.ProviderResult<languages.CompletionList> => {
+    switch (wordContext) {
+      case '':
+        return this.getCompletionsFromDefinitions(getGlobalSuggestions(this.monaco), getAlertSuggestions(this.monaco));
+      case 'Alerts':
+        return this.getCompletionsFromDefinitions(getAlertsSuggestions(this.monaco));
+      case 'GroupLabels':
+      case 'CommonLabels':
+      case 'CommonAnnotations':
+      case 'Labels':
+      case 'Annotations':
+        return this.getCompletionsFromDefinitions(getKeyValueSuggestions(this.monaco));
+      default:
+        return { suggestions: [] };
+    }
+  };
+
+  private getCompletionsFromDefinitions = (...args: SuggestionDefinition[][]): languages.CompletionList => {
+    const allDefinitions = concat(...args);
+
+    return {
+      suggestions: allDefinitions.map((definition) => buildAutocompleteSuggestion(definition, this.range)),
+    };
+  };
+}
+
+function buildAutocompleteSuggestion(
+  { label, detail, documentation, kind, insertText }: SuggestionDefinition,
+  range: IRange
+): languages.CompletionItem {
+  const insertFallback = typeof label === 'string' ? label : label.label;
+  const labelObject = typeof label === 'string' ? { label: label, description: detail } : { ...label };
+
+  labelObject.description ??= detail;
```

Comment:
I've never really seen a good use-case for logical nullish assignments, why do we want to assign `description` only if it's `null` here?

Reply:
I want to use `detail` only as a fallback value if `description` doesn't exist.
Suggestions looks better in the editor if configured this way